addons/l10n_latam_check/models/account_payment.py

```diff
@@ -245,9 +256,10 @@ def _prepare_move_line_default_vals(self, write_off_line_vals=None, force_balanc
         # we dont check the payment method code because when deposited on bank/cash journals pay method is manual but we still change the label
         # we dont want this names on the own checks because it doesn't add value, already each split/check line will have it name
         elif (self.l10n_latam_new_check_ids or self.l10n_latam_move_check_ids) and self.payment_method_code != 'own_checks':
+            check_name = [check_name for check_name in (self.l10n_latam_new_check_ids | self.l10n_latam_move_check_ids).mapped('name') if check_name]
             document_name = (
                 _('Checks %s received') if self.payment_type == 'inbound' else _('Checks %s delivered')) % (
-                ', '.join((self.l10n_latam_new_check_ids or self.l10n_latam_move_check_ids).mapped('name'))
+                ', '.join(check_name)
             )
```

Comment:
Bug fix to be able to have a payment with check without number otherwise it traceback (need to be backported)